Feature/mapping

yamadayy opened the pull request.

YoshiyukiYamada committed:
Add new classes for mapping study.

YoshiyukiYamada committed:
Add parameters.py and test_mission.py. Also add pyshon codes and data files which used in parameters.py.

YoshiyukiYamada committed:
Add modules used in the codes to requirements.txt for avoiding the error in github action.

YoshiyukiYamada committed:
Add empty methods.

YoshiyukiYamada committed:
Add empty methods.

YoshiyukiYamada committed:
Apply syntax of Python 3.11 for efficiency.py
Fix typo in mapping.py
Values of detector_separation_x and y are fixed in parameters.py
Use fixed values in parameters.py for calculating gap in pointing_plan_factory.py.

YoshiyukiYamada committed:
Subdirectory attitude under the attitude is stupid. Move attitude.py to satellite/attitude and remove directory satellite/attitude/attitude.
imporrt sentence in satellite/satellite.py is fixed.

YoshiyukiYamada committed:
Add argument orbit in constructor of Attitude in attitude.py.
Add new enum EnumPointingFreedom ti pointing_plan_factory.py.
Modify constructor of Satellite in satellite.py.

YoshiyukiYamada committed:
attitude.py is under construction.
Add astropy to requirements.txt.

YoshiyukiYamada committed:
Implement method get_position_angle in Attitude class.

YoshiyukiYamada committed:
Add TODO

YoshiyukiYamada committed:
Two enum classes coded in pointing_plan_factory.py are moved to separate files because it causes error of circular import.
Import sentences in some files which import Enum classes are fixed.
Argument of constructor of Satellite is changed. Mode is needed because Attitude class is not visible from PointingPlanFactory.

YoshiyukiYamada committed:
Fix import sentences and related errors.

YoshiyukiYamada committed:
Add ltan to parameters.py.
orbit.py uses ltan and inclination.

YoshiyukiYamada committed:
Member attributes are made to be private in attitude.py, mapping.py, orbit.py, pointing_plan_factory.py, pointing_plan.py, and satellite.py.
Fix warning in efficiency.py.

YoshiyukiYamada committed:
The method create in pointing_plan_factory.py is not property. Remove @property line.
EQUATORIAL_EARTH_RADIUS is made to be property in parameters.py.
The file orbit.py is under construction.

YoshiyukiYamada committed:
under construction

YoshiyukiYamada committed:
Implement class Orbit. Fix formula used in satellite_direction and observable_p.

YoshiyukiYamada committed:
An argument orbit is added in constructor of Attitude.
An argument initial_time is added in constructor of Satellite.
Four parameters max and min of l and b are added.

YoshiyukiYamada committed:
Add initial_time as an argument of constructor of Mapping.
The method included_p is added to class Mapping.

YoshiyukiYamada committed:
Move test code for included_p from mapping.py to test_mapping.py.

e-yamane committed:
Confirmation that the test works for now.

YoshiyukiYamada committed:
Add type hints of class attributes in constructor.
implement get_position_angle in pointing_plan_factory.py and satellite.py.

YoshiyukiYamada committed:
Merge remote-tracking branch 'origin/feature/mapping' into feature/mapping

YoshiyukiYamada committed:
quick fix. remove calling get_position_angle in pointing_plan_factory.py.

YoshiyukiYamada committed:
add comment (TODO)

e-yamane committed:
implements dirty flag

e-yamane committed:
Merge branch 'feature/mapping' of github.com:JASMINE-Mission/jasmine_toolkit into feature/mapping

YoshiyukiYamada committed:
Functions other than create is made to be protected in pointing_plan_factory.py. Add some TODOs.
The function vstack does not change the array in argument itself. Need to copy return value (array) to itself. Also return value of get_array should be copy. (in pointing_plan.py)
The function satellite_direction is divided into 4 functions. The function name of observable_p is changed to is_observable.(orbit.py)

YoshiyukiYamada committed:
Numpy copy function automatically decide the copy should be deep or shallow copy, and it is not controllable from the user. So, change array.copy() to copy.deepcopy(array) as return value of get_array function. (pointing_plan.py)

YoshiyukiYamada committed:
A test of previous commit failed. So, roll back and add TODO comment.

YoshiyukiYamada committed:
A test of previous commit failed. So, roll back and add TODO comment.

e-yamane committed:
Make the dirty/clean decision in one place

e-yamane committed:
Implementation of parameters2.py (in progress)

e-yamane committed:
Implementation of parameters2.py (roughly finished)

YoshiyukiYamada committed:
Japanese character causes error in command line pytest of windows. Description of EARTH_MASS is changed to Earth mass (English).

e-yamane committed:
Specify the encoding of the read file

e-yamane committed:
Set variables in parameters.py to constants.yaml

e-yamane committed:
Confirmation that it works by importing properties defined in parameters.py.

YoshiyukiYamada committed:
grid, n_l, n_b are made to be local variables from class attributes

YoshiyukiYamada committed:
Remove argument self because it is static method.

YoshiyukiYamada committed:
Remove print sentence for debug.

YoshiyukiYamada committed:
Reformat attitude.py and mapping.py.
Argumane of PointingPlanFactory is added (mapping.py)
Remove attribute __parameter. Use Parameters.getInstance() instead.
Add attribute __gap_on_the_sky.
Return type of _find_next_position is changed to SkyCoord.

e-yamane committed:
Resolves static analysis warnings

e-yamane committed:
Resolves static analysis warnings

YoshiyukiYamada committed:
Reformatting test_mapping.py
Add next_observable_time method in satellite.py
Implement _generate_observation_time method in pointing_plan_factory.py.
Reformat file orbit.py.

YoshiyukiYamada committed:
Merge remote-tracking branch 'origin/feature/mapping' into feature/mapping

# Conflicts:
#	jasmine_toolkit/utils/test/test_parameters2.py

YoshiyukiYamada committed:
For avoiding erfa warning, using warning method is dangerous. So, when we use time after 2027, explicitly specify "scale = "tcb"".

YoshiyukiYamada committed:
Move included_p method from mapping.py to pointing_plan.py.
test_mapping.py is fixed.
Simplify local variable names in mapping.py
Move some methods from pointing_plan_factory.py to pointing_plan.py.
move method test_included_p from test_mapping.py to test_pointing_plan.py.
The method test_create is temporally implemented in test_mapping.py.
Reformat parameters.py

YoshiyukiYamada committed:
The unit of return value of method get_position_angle is made to be radian from degree. (attitude.py)
Add class attribute grid_coord which hold coordinate of grid in icrs, and set its value within the method _generate_grid in pointing_plan.py.
In the method make_observation, rever grid_coord other than target(pointing) in pointing_plan.py.
Sometimes the argument of math.asin is become larger than 1 (very small value due to calculation error.) For avoiding math domain error, round it to -1.0 ~ 1.0.
Argument name pa is used also as local variable name. Change name.

YoshiyukiYamada committed:
Add enum class (fov_change_mode.py).
Under construction for modifying number of observation be realistic.

YoshiyukiYamada committed:
Change variable and faunction names.(pointing_plan_factory.py)

e-yamane committed:
Try a format where you work from a specific module and not from yaml.

YoshiyukiYamada committed:
In pointing_plan_factory.py:
The definition of self.__time_per_a_fov is changed.
Make array for time sequence of observation first, and separate loop of calculating pointing.
The methods _find_next_observation_time and _get_position angle only call the methods in satellite. So remove these methods and call methods in satellite class directory.
In satellite.py:
Add new method observation_count which counts available number of observation in small frame.
Add self.__target which means rough observation direction.

YoshiyukiYamada committed:
pointing_plan.py:
Add methods _coord_to_grid and _grid_to_coord.
Definition of fov_x and fov_y is moved from method _get_field_of_view to constructor.
Update create function in pointing_plan_factory.py.

YoshiyukiYamada committed:
Small maneuver pointing change is implemented in the method create in pointing_plan_factory.py.
The assignment of __gap_on_the_sky in pointing_plan.py is moved from _generate_grid to constructor.
Add method pointing_by_small_maneuver is added to pointing_plan.py.

YoshiyukiYamada committed:
Use fixture for reducing computation time.
The code written in if-main block is moved to test. (pointing_plan.py)
Remove unused import (attitude.py)
Add test for attitude.py.

YoshiyukiYamada committed:
Remove if name=main block in mapping.py and pointing_plan_factory.py.

YoshiyukiYamada committed:
Add empty file test_orbit.py and test_satellite.py
The method calc_statistics is under construction in mapping.py and test_mapping.py.

YoshiyukiYamada committed:
Implement calc_statistics in mapping.py and its test in test_mapping.py.

YoshiyukiYamada committed:
Apply comments by code climate.

e-yamane committed:
Make an expression hold as an expression

YoshiyukiYamada committed:
mapping.py and test_mapping.py: Output of calc_statistics is fixed.
pointing_plan.py and test_pointing_plan_factory.py: method name get_array is changed to get_grid. An attribute __plan is added. Methods save_grid, save_plan, get_plan are added. (temporally test codes for append ndarray is added as main)

YoshiyukiYamada committed:
Add save methods.

YoshiyukiYamada committed:
mapping.py: if A^T dot A is singular, skip calculation of parameter dispersion.

YoshiyukiYamada committed:
Add condition of solar aspect angle to next_observation_time in orbit.py.
In mapping.py, change variable name of main block for avoiding conflict of the variable in class method.
Also output number of observation in one small field in mapping.py.

YoshiyukiYamada committed:
Fix condition of solar aspect angle.

YoshiyukiYamada committed:
pointing_plan.py:
Multiple loop to one line (save_grid).
Simplify pointing_by_small_maneuver, included_p.
pointing_plan_factory.py:
separate create method for avoiding cognitive complexity.
mapping.py, orbit.py, and satellite.py:
Reformat files.

YoshiyukiYamada committed:
Reformat file

YoshiyukiYamada committed:
Fix condition of solar aspect angle.

YoshiyukiYamada committed:
Merge remote-tracking branch 'origin/feature/mapping' into feature/mapping

e-yamane committed:
認知的複雑度緩和

e-yamane committed:
もうちょっと認知負荷下げる

YoshiyukiYamada committed:
mapping.pyの気持ち悪いのを変更。
pointing_play, code climate警告の対応。

YoshiyukiYamada committed:
pointing_play, code climate警告の対応。

e-yamane committed:
parameters.pyにある定義を（ほぼ）移してきた。average_telescope_throughputと、average_quantum_efficiencyと、total_efficiencyはエラーがでます。

e-yamane committed:
CodeClimateの指摘対応

e-yamane committed:
CodeClimate対応

e-yamane committed:
CodeClimate指摘対応

e-yamane committed:
CodeClimate対応

e-yamane committed:
CodeClimate指摘対応

e-yamane committed:
CodeClimate指摘対応(?)

e-yamane committed:
CodeClimate指摘対応

e-yamane committed:
CodeClimate対応+JasmineConstantで発生していたエラーに対する対応

e-yamane committed:
CodeClimate対応

e-yamane committed:
CodeClimate対応

e-yamane committed:
CodeClimate対応

e-yamane committed:
一旦Quantity形式で値を持たないようにした

e-yamane committed:
test_mission.pyのテストをtest_parameters2.pyに移動

e-yamane committed:
Parameters -> Parameters2に差し替えて全件テストがパスした

e-yamane committed:
Parameters#get_instance経由でParameters2を取得するのを辞めて、Parametersを完全削除

e-yamane committed:
Paramers2 -> Parametersへ名称変更

e-yamane committed:
yaml形式の値は不要なので削除(yaml形式のimport方法についてはテストがある）

e-yamane committed:
Paramersの一部に単位を付与した

e-yamane committed:
Merge pull request #4 from JASMINE-Mission/feature/switch_parameter

Feature/switch parameter

e-yamane committed:
山田さんのフィードバック対応

e-yamane committed:
エラーが起こらないところまで

e-yamane committed:
エラーが出たので出なくなるまで進める

e-yamane committed:
エラーが出ないところまで進める

e-yamane committed:
test_formulaに含んでいるので消す

e-yamane committed:
単位を捨てないと通し方が分からないところ

e-yamane committed:
efficiencyのevaluteメソッドで単位を捨てなくても動くようにした

e-yamane committed:
orbitのradを残すようにしてみた

e-yamane committed:
エラーが出ないところまで進める

e-yamane committed:
earth_avoiding_angleに単位を付与する

e-yamane committed:
単位付与全部終了

YoshiyukiYamada committed:
Apply codeclimate comments.

YoshiyukiYamada committed:
Apply codeclimate comments.

YoshiyukiYamada committed:
Add tests for attityde.py

YoshiyukiYamada committed:
Add tests for attityde.py

YoshiyukiYamada committed:
Fix tests for attityde.py

astronasutarou commented on jasmine_toolkit/satellite/attitude/__init__.py:
```diff
@@ -0,0 +1,2 @@
+class Attitude:
+    pass
```
この記述は不必要かと思いましたが残したほうが良い理由があるでしょうか．

yamadayy replied:
済みません。__init__.pyがいつどう編集されるのかはあくしておらず、知らないうちに変更になっていたようです。消しました。

astronasutarou commented on jasmine_toolkit/satellite/attitude/attitude.py:
```diff
@@ -0,0 +1,50 @@
+            if ra >= math.pi:
+                ra_np = ra - math.pi
+            else:
+                ra_np = ra + math.pi
```
pointing から真北の方向に 90° 回転させた方向を計算しているという理解で良いでしょうか？ `directional_offset_by` という method があるのでそちらを使うのはどうでしょう．角度範囲を気にする必要がないですし `SkyCoord` が返ってくるので `position_angle` method が使えます．

yamadayy replied:
書き直しました。

astronasutarou commented on jasmine_toolkit/utils/mapping.py:
```diff
@@ -0,0 +1,80 @@
+        pass
+
+    @staticmethod
+    def make_data(freedom: EnumPointingFreedom, mode: EnumPointingMode,
+                  initial_time: Time, duration: TimeDelta):
```
`make_data` は関数名として情報がなさすぎるかと思いました．

yamadayy replied:
make_pointing_planとしました。

astronasutarou commented on jasmine_toolkit/utils/mapping.py:
```diff
@@ -0,0 +1,80 @@
+        return pointing_plan
+
+    @staticmethod
+    def calc_statistics(tc: Time, data: list):
```
`data`, `answer`, `matrix_a` あたりの変数名が一般的すぎて何を意味するのかわかりにくい (可読性・メンテナンス性が低い) と感じました．また精度計算の部分は別の関数に切り出したほうがわかりやすそうです (Mapping が持つべき関数なのかは検討の余地があるかと思いますが)．

yamadayy replied:
dataとanserについては対応しました。matrix_aは論文のAという行列なので、その旨コメントを加えました。

astronasutarou reviewed (approved):
計算している部分についてはほぼチェックしていません．流し読みして特に目に付いたところだけコメントしました．対応していただかなくても構いませんが `/satellite/attitude/__init__.py` に空の `Attitude` が残っている件は対処しておいたほうがいいかと思います．

YoshiyukiYamada committed:
Apply pull request review comments.
Remove class Attitude in __init__.py.
Codes in get_attitude method is rewritten by using directional_offset_by method in astropy.
Variable names data and answer are changed to mapping result and statistical results. Add comment of variable name matrix_a.

astronasutarou reviewed (approved):
修正内容を確認しました．
